design/src/rfcs/rfc0024_request_id.md

```diff
@@ -0,0 +1,136 @@
+RFC: RequestID in business logic handlers
+=============
+
+> Status: RFC
+>
+> Applies to: server
+
+For a summarized list of proposed changes, see the [Changes Checklist](#changes-checklist) section.
+
+Terminology
+-----------
+
+- **RequestID**: a service-wide unique identifier
+- **UUID**: a universally unique identifier
+
+RequestID is an element that uniquely identifies a client request. RequestID is used by services to map all logs, events and
+specific data to a single operation. This RFC discusses whether and how smithy-rs can make that value available to customers.
+
+Services use a RequestID to collect logs related to the same request and see its flow through the various operations,
+help clients debug requests by sharing this value and, in some cases, use this value to perform their business logic. RequestID is unique across a service at least within a certain timeframe.
+
+This value for the purposes above must be set by the service. Clients could send any value not conforming to the valid, chosen format,
+avoid sending one altogether or exploit how some service is using that value
+(like in [1](https://en.wikipedia.org/wiki/Shellshock_(software_bug)) and
+[2](https://cwiki.apache.org/confluence/display/WW/S2-045)).
```

Comment:
The sentence about the client sending the Request Id seems to be pretty much the same as the list directly below?